src/dmsort.rs

```diff
@@ -317,8 +320,8 @@ where
 						let old_len = s.dropped.len();
 						s.dropped.reserve(num_backtracked);
 						ptr::copy_nonoverlapping(
-							s.slice.get_unchecked(s.write),
-							s.dropped.get_unchecked_mut(old_len),
+							s.slice.as_ptr().add(s.write),
+							s.dropped.as_mut_ptr().add(old_len),
```

Comment:
Both of these indexes are out-of-bounds for the slice, but inbounds for the backing allocation.